internal/runtime/init/init_test.go

```diff
@@ -183,6 +188,239 @@ p = true { 1 = 2 }`
+		{
+			note:          "load multiple bundles. one tar.gz and one folder. Bundle mode is true",
+			folderContent: bundle1Folder,
+			bundleInfoTC: []bundleInfo{
+				bundle1TarGz,
+				bundle1FolderInfo,
+			},
+			expectedBundles: 2,
+			expectedModules: 0,
+			asBundle:        true,
+		},
+		{
+			note:          "load multiple bundles. one tar.gz and one folder. Bundle mode is false",
+			folderContent: bundle1Folder,
+			bundleInfoTC: []bundleInfo{
+				bundle1TarGz,
+				bundle1FolderInfo,
+			},
+			expectedBundles: 1,
+			expectedModules: 1,
+			asBundle:        false,
+		},
+		{
+			note:          "load multiple bundles. two tar.gz and one folder. Bundle mode is true",
+			folderContent: bundle1Folder,
+			bundleInfoTC: []bundleInfo{
+				bundle1TarGz,
+				bundle2TarGz,
+				bundle1FolderInfo,
+			},
+			expectedBundles: 3,
+			expectedModules: 0,
+			asBundle:        true,
+		},
+		{
+			note:          "load multiple bundles. two tar.gz and one folder. Bundle mode is false",
+			folderContent: bundle1Folder,
+			bundleInfoTC: []bundleInfo{
+				bundle1TarGz,
+				bundle2TarGz,
+				bundle1FolderInfo,
+			},
+			expectedBundles: 2,
+			expectedModules: 1,
+			asBundle:        false,
+		},
+		{
+			note:          "load just one folder. Bundle mode is true",
+			folderContent: bundle1Folder,
+			bundleInfoTC: []bundleInfo{
+				bundle1FolderInfo,
+			},
+			expectedBundles: 1,
+			expectedModules: 0,
+			asBundle:        true,
+		},
+		{
+			note:          "load just one folder. Bundle mode is false",
+			folderContent: bundle1Folder,
+			bundleInfoTC: []bundleInfo{
+				bundle1FolderInfo,
+			},
+			expectedBundles: 0,
+			expectedModules: 1,
+			asBundle:        false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.note, func(t *testing.T) {
+
+			test.WithTempFS(tc.folderContent, func(rootDir string) {
+				paths := []string{}
+				for _, bdlInfo := range tc.bundleInfoTC {
+					if strings.HasSuffix(bdlInfo.fileName, ".tar.gz") {
+
+						// Create the tar gz files temporarily
+						buf := archive.MustWriteTarGz(bdlInfo.files)
+						bundleFile := filepath.Join(rootDir, bdlInfo.fileName)
+						out, err := os.Create(bundleFile)
+						if err != nil {
+							t.Fatalf("Unexpected error: %v", err)
+						}
+						_, err = out.Write(buf.Bytes())
+						if err != nil {
+							t.Fatalf("Unexpected error: %v", err)
+						}
+					}
+
+					paths = append(paths, filepath.Join(rootDir, bdlInfo.fileName))
+				}
+
+				loaded, err := LoadPaths(paths, nil, tc.asBundle, nil, true, false, nil, nil)
+				if err != nil {
+					t.Fatal("Failed LoadPaths ", err)
+				}
+				if tc.expectedBundles != len(loaded.Bundles) {
+					t.Fatalf("Expected %d bundles, got %d", tc.expectedBundles, len(loaded.Bundles))
+				}
+				if tc.expectedModules != len(loaded.Files.Modules) {
+					t.Fatalf("Expected %d modules, got %d", tc.expectedModules, len(loaded.Files.Modules))
+				}
+
+				// Testing the content
+				for path, actual := range loaded.Bundles {
```

Comment:
I think here in order to test the content it'd be best to ensure that the value at particular paths is correct e.g. `data.a.foo1` is set to `bar2` as we expect. This way we can be sure that not only the lengths of the modules is correct but also the contents of them. Hope that makes sense! 😊

Reply:
Ahh, sorry. I think I'd misread this. What we have looks fine to me. Sorry for the confusion.